apps/web/src/pages/Communities/CommunityMembersPage/components/RemoveCommunityMemberDialog.tsx

```diff
@@ -0,0 +1,152 @@
+import { useState } from "react";
+import { useIntl } from "react-intl";
+import TrashIcon from "@heroicons/react/20/solid/TrashIcon";
+import { useMutation } from "urql";
+import { useOutletContext } from "react-router-dom";
+
+import { Dialog, Button, Chip, Chips } from "@gc-digital-talent/ui";
+import {
+  commonMessages,
+  formMessages,
+  getLocalizedName,
+} from "@gc-digital-talent/i18n";
+import { toast } from "@gc-digital-talent/toast";
+import {
+  RoleInput,
+  CommunityMembersPage_CommunityFragment as CommunityMembersPageCommunityFragmentType,
+} from "@gc-digital-talent/graphql";
+
+import { getFullNameLabel } from "~/utils/nameUtils";
+import { CommunityMember } from "~/utils/communityUtils";
+
+import { UpdateUserCommunityRoles_Mutation } from "./operations";
+import { ContextType } from "./types";
+
+interface RemoveCommunityMemberDialogProps {
+  user: CommunityMember;
+  community: CommunityMembersPageCommunityFragmentType;
+}
+
+const RemoveCommunityMemberDialog = ({
+  user,
+  community,
+}: RemoveCommunityMemberDialogProps) => {
+  const intl = useIntl();
+  const { teamId } = useOutletContext<ContextType>();
+  const [{ fetching }, executeMutation] = useMutation(
+    UpdateUserCommunityRoles_Mutation,
+  );
+  const [isOpen, setIsOpen] = useState<boolean>(false);
+  const roleInputArray: RoleInput[] = user.roles.map((role) => {
+    return { roleId: role.id, teamId };
+  });
+
+  const handleRemove = async () => {
+    await executeMutation({
+      updateUserRolesInput: {
+        userId: user.id,
+        roleAssignmentsInput: {
+          detach: roleInputArray,
+        },
+      },
+    })
+      .then((res) => {
+        if (!res.error) {
+          setIsOpen(false);
+          toast.success(
+            intl.formatMessage({
+              defaultMessage: "Member removed successfully",
+              id: "KWXIKL",
+              description:
+                "Alert displayed to user when a community member is removed",
+            }),
+          );
+        }
+      })
+      .catch(() => {
+        toast.error(
+          intl.formatMessage({
+            defaultMessage: "Member removal failed",
+            id: "fsvf1L",
+            description:
+              "Alert displayed to user when an error occurs while removing a community member",
+          }),
+        );
+      });
```

Comment:
Pretty sure we have this pattern else where but, due to graphql, we can get an error without anything being thrown on the request so we may need to move the error toast into a different funcion so we can also pop it on an error that is not thrown.

```tsx
const handleError = () => { 
  // .. toast.error
  }
  
  executeMutatioin({ ... })
     .then((res) => {
        if(!res.error) {
           // happy path
        } else {
          handleError();
        } 
     }).catch(handleError);
 ```

Reply:
2349445afebd31b147708a569a1b4dfe6aa010b7